From 31614bf78d8d5740fc4b38385de2d69cc04dffa9 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Wed, 16 Jul 2014 11:38:12 +0100
Subject: [PATCH] MusicPlayer: On stop clear any temporary one shot metadata

This fixes a problem getting stuck playing the same track after exiting the
import screen if a track was test played while on there.
---
 mythplugins/mythmusic/mythmusic/importmusic.cpp |    2 +-
 mythplugins/mythmusic/mythmusic/musicplayer.cpp |    6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/mythplugins/mythmusic/mythmusic/importmusic.cpp b/mythplugins/mythmusic/mythmusic/importmusic.cpp
index 8ccda16..4c465cf 100644
--- a/mythplugins/mythmusic/mythmusic/importmusic.cpp
+++ b/mythplugins/mythmusic/mythmusic/importmusic.cpp
@@ -107,7 +107,7 @@ ImportMusicDialog::~ImportMusicDialog()
     if (gPlayer->getCurrentMetadata() && m_playingMetaData)
     {
         if (gPlayer->isPlaying() && gPlayer->getCurrentMetadata()->Filename() == m_playingMetaData->Filename())
-            gPlayer->stop();
+            gPlayer->stop(true);
     }
 
     if (m_locationEdit)
diff --git a/mythplugins/mythmusic/mythmusic/musicplayer.cpp b/mythplugins/mythmusic/mythmusic/musicplayer.cpp
index fdf777e..00c064f 100644
--- a/mythplugins/mythmusic/mythmusic/musicplayer.cpp
+++ b/mythplugins/mythmusic/mythmusic/musicplayer.cpp
@@ -274,6 +274,12 @@ void MusicPlayer::stop(bool stopAll)
         m_output->Reset();
     }
 
+    if (m_oneshotMetadata)
+    {
+        delete m_oneshotMetadata;
+        m_oneshotMetadata = NULL;
+    }
+
     m_isPlaying = false;
 
     if (stopAll && getDecoder())
-- 
1.7.10.2

